src/Chapter2.hs

```diff
@@ -336,7 +348,9 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList s e l
+  | s < 0 || e < 0 = []
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding a separate case to your check :slightly_smiling_face:

Reply:
Thank you for comment. But, unfortunately, I'm not sure how to implement it more efficient. What separate case should I add?

Reply:
By having a separate check I mean just adding a separate condition to this guard `|| s > e`

Reply:
Ah! If e is large, it will take time to calculate take (e + 1). But its value is unnecessary if s > e. So, it is not efficient. So, you wrote it can be more efficient. I've got it. Thank you. I revised the code.